packages/core/src/actions/getTransactions.ts

```diff
@@ -0,0 +1,44 @@
+import { TransactionsFetchError } from '../errors/transaction.js'
+import type { UTXOSchema } from '../transports/types.js'
+import type { Account } from '../types/account.js'
+import type { Chain } from '../types/chain.js'
+import type { Client } from '../types/client.js'
+import type { UTXOTransaction } from '../types/transaction.js'
+import type { Transport } from '../types/transport.js'
+
+export type GetTransactionsParameters = {
+  address: string
+  offset?: number
+  limit?: number
+  lastBlock?: string
+  afterTxId?: string
+}
+
+export type GetTransactionsReturnType = {
+  transactions: Array<Partial<UTXOTransaction>>
+  total: number
+  itemsPerPage: number
+  hasMore: boolean
+}
+
+export async function getTransactions<
+  C extends Chain | undefined,
+  A extends Account | undefined = Account | undefined,
+>(
+  client: Client<Transport, C, A, UTXOSchema>,
+  { address, offset, limit, lastBlock }: GetTransactionsParameters
+): Promise<GetTransactionsReturnType> {
+  try {
+    return client.request({
+      method: 'getTransactions',
+      params: {
+        address,
+        offset,
+        limit,
+        lastBlock,
+      },
+    })
+  } catch (_error) {
+    throw new TransactionsFetchError({ address })
```

Comment:
Similarly here we should try to include the original error in the TransactionsFetchError